alpha/alpha-fsm/src/main/java/org/apache/servicecomb/pack/alpha/fsm/SagaShardRegionActor.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.servicecomb.pack.alpha.fsm;
+
+import akka.actor.AbstractActor;
+import akka.actor.ActorRef;
+import akka.actor.ActorSystem;
+import akka.actor.Props;
+import akka.cluster.sharding.ClusterSharding;
+import akka.cluster.sharding.ClusterShardingSettings;
+import akka.cluster.sharding.ShardRegion;
+import java.lang.invoke.MethodHandles;
+import org.apache.servicecomb.pack.alpha.core.fsm.event.base.BaseEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SagaShardRegionActor extends AbstractActor {
+
+  private static final Logger LOG = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  private final ActorRef sagaActorRegion;
+
+  static ShardRegion.MessageExtractor messageExtractor = new ShardRegion.MessageExtractor() {
+    @Override
+    public String entityId(Object message) {
+      if (message instanceof BaseEvent) {
+        return ((BaseEvent) message).getGlobalTxId();
+      } else {
+        return null;
+      }
+    }
+
+    @Override
+    public Object entityMessage(Object message) {
+      return message;
+    }
+
+    @Override
+    public String shardId(Object message) {
+      int numberOfShards = 10; // NOTE: Greater than the number of alpha nodes
```

Comment:
This number should be passed as a parameter, do we need to know the alpha instance number at this time?

Reply:
> This number should be passed as a parameter, do we need to know the alpha instance number at this time?

I feel that 10 nodes are enough, maybe later we can modify it to be user-configurable